### ISSUE TITLE
reproject_exact() and reproject_interp() default arguments inconsistent

### ISSUE THREAD

**Issue by mhardcastle** [astrofrog/reproject#118]
reproject_interp() defaults to hdu_in=0, but reproject_exact() defaults to hdu_in=None. This means that working code where reproject_interp() is passed a hdulist fails when reproject_exact() is used instead with the same arguments; moreover it fails with a very obscure error. Unless there's a good reason I am missing, suggest that the default be changed in reproject_exact().

**Comment by bsipocz** [comment]
Indeed, the default of ``reproject_interp()`` has been changed in #97 as part of generalizing cube reprojection that was released in v0.3. That work was done for the interpolations but not for the polygon intersection. I can't see a reason not to change None to 0 there, too.
What do you say @keflavich?

**Comment by keflavich** [comment]
Agreed, `hdu_in=0` is a reasonable default.

**Comment by bsipocz** [comment]
Shall I open a PR, or would you push directly to master?

**Comment by keflavich** [comment]
I'll put a PR here.  But first, @mhardcastle, what was the "obscure error"?  It should have been:
`"More than one HDU is present, please specify HDU to use with ``hdu_in=`` option"`
which I'd say is not so obscure.  If something else showed up, that indicates a bug.

**Comment by mhardcastle** [comment]
'Obscure error' as follows:

```
Traceback (most recent call last):
  File "/home/mjh/git/ddf-pipeline/mosaic.py", line 96, in <module>
    r, footprint = reproject_exact(hdus[i], header, parallel=True)
  File "/usr/lib64/python2.7/site-packages/reproject-0.4.dev366-py2.7-linux-x86_64.egg/reproject/spherical_intersect/high_level.py", line 56, in reproject_exact
    array_in, wcs_in = parse_input_data(input_data, hdu_in=hdu_in)
  File "/usr/lib64/python2.7/site-packages/reproject-0.4.dev366-py2.7-linux-x86_64.egg/reproject/utils.py", line 19, in parse_input_data
    return parse_input_data(input_data[hdu_in])
  File "/soft/python/lib64/python2.7/site-packages/astropy/io/fits/hdu/hdulist.py", line 196, in __getitem__
    idx = self.index_of(key)
  File "/soft/python/lib64/python2.7/site-packages/astropy/io/fits/hdu/hdulist.py", line 519, in index_of
    raise KeyError(key)
KeyError: None
```

To be clear, hdus[i] in the original call is a hdulist, one of a list of hdulists to be mosaiced. The error message is what you would expect if the code indexes the hdulist with None, but it's hardly user-friendly...

**Comment by keflavich** [comment]
Ah, this is actually a bug.  I'll fix this error message in #119 too

**Comment by astrofrog** [comment]
Fixed in #119
